executor/grant.go

```diff
@@ -123,44 +123,44 @@ func (e *GrantExec) Close() error {
 
 // Check if DB scope privilege entry exists in mysql.DB.
 // If unexists, insert a new one.
-func (e *GrantExec) checkAndInitDBPriv(user string, host string) error {
-	db, err := e.getTargetSchema()
+func checkAndInitDBPriv(ctx context.Context, dbName string, is infoschema.InfoSchema, user string, host string) error {
+	db, err := getTargetSchema(ctx, dbName, is)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	ok, err := dbUserExists(e.ctx, user, host, db.Name.O)
+	ok, err := dbUserExists(ctx, user, host, db.Name.O)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	if ok {
 		return nil
 	}
 	// Entry does not exist for user-host-db. Insert a new entry.
-	return initDBPrivEntry(e.ctx, user, host, db.Name.O)
+	return initDBPrivEntry(ctx, user, host, db.Name.O)
 }
 
 // Check if table scope privilege entry exists in mysql.Tables_priv.
 // If unexists, insert a new one.
-func (e *GrantExec) checkAndInitTablePriv(user string, host string) error {
-	db, tbl, err := e.getTargetSchemaAndTable()
+func checkAndInitTablePriv(ctx context.Context, dbName, tableName string, is infoschema.InfoSchema, user string, host string) error {
```

Comment:
There many repeat code now, I've thought this problem.
But I found that `GrantExec` and `RevokeExec` behavior differently, I decide not to merge them to reuse code.